pkg/apiserver/api_installer.go

```diff
@@ -730,6 +695,9 @@ func (n scopeNaming) GenerateLink(req *restful.Request, obj runtime.Object) (pat
 			return "", "", err
 		}
 	}
+	if len(name) == 0 {
+		return "", "", errEmptyName
+	}
```

Comment:
@smarterclayton can you take a look at this? `rootScopeNaming#GenerateLink` returned an err if Name was empty, and `TestSelfLinkSkipsEmptyName` was complaining unless scopeNaming#GenerateLink behaved similarly